akka-javasdk-tests/src/test/java/akkajavasdk/components/workflowentities/TransferWorkflow.java

```diff
@@ -68,29 +68,31 @@ private StepEffect withdraw() {
     var state = currentState().withLastStep("withdrawn").asAccepted();
     return stepEffects()
         .updateState(state)
-        .transitionTo("deposit"); // FIXME: use method ref and make it typed
+        .thenCall(TransferWorkflow::deposit)
+        .withInput(new Deposit(currentState().transfer().to(), currentState().transfer().amount()));
   }
 
-  private StepEffect deposit() {
+
+  private StepEffect deposit(Deposit deposit) {
 
     var toWallet = currentState().transfer().to();
     var amount = currentState().transfer().amount();
 
     componentClient
-      .forKeyValueEntity(toWallet)
-      .method(WalletEntity::deposit)
-      .invoke(amount);
+        .forKeyValueEntity(toWallet)
+        .method(WalletEntity::deposit)
+        .invoke(amount);
 
     var state = currentState().withLastStep("deposited").asFinished();
     return stepEffects()
       .updateState(state)
-      .transitionTo("logAndStop"); // FIXME: use method ref and make it typed
+      .thenCall(TransferWorkflow::logAndStop);
```

Comment:
StepEffects builder closed with a `thenX` method. This aligns with the other effect builders we have. 

Could be `thenTransitionTo`, but like argued above, if more natural if we call it `call`.

WDYT?

Reply:
I like `thenTransitionTo` more, because `call, calling` is already heavily used in many other aspects, like component client, HTTP endpoint etc. I think it will be good to keep a separate term for switching the step.

Reply:
Fine, let's stick with `transition`. 

I also got that feeling that `call` + `thenReply` could be misleading. Good point.